src/raygui.h

```diff
@@ -722,6 +722,7 @@ RAYGUIAPI int GuiComboBox(Rectangle bounds, const char *text, int *active);
 RAYGUIAPI int GuiDropdownBox(Rectangle bounds, const char *text, int *active, bool editMode);          // Dropdown Box control, returns selected item
 RAYGUIAPI int GuiSpinner(Rectangle bounds, const char *text, int *value, int minValue, int maxValue, bool editMode); // Spinner control, returns selected value
 RAYGUIAPI int GuiValueBox(Rectangle bounds, const char *text, int *value, int minValue, int maxValue, bool editMode); // Value Box control, updates input text with numbers
+RAYGUIAPI int GuiValueBoxF(Rectangle bounds,const char* text, char *val_str, float *value, bool editMode); // Floating point Value box control, val_str is buffer where string representation is stored
```

Comment:
Why is `char *val_str` required? `float *value` should be enough to be equivalent to `GuiValueBox()` function.

Reply:
Hey,
i implemented it this way because when you print out a float without a format specifier you don't know how many digits it will take and when you provide a format specifier for digits after comma, you can never delete a digit because the float is printed with the same number of digits after you delete. For an integer there is only a single way to print it out.

Reply:
I see and it's useful but it would be nice that it worked like `printf()` format parameter so the user could specify something like `03.3` or `2'3` or `4,02`, defining the desired integer-decimal digits, if padded with zeroes and even the format for the decimal separator. Would it be possible to implement that?

Additionally, it should be renamed to `char *valueFormat` and moved after `float *value` parameter.

Reply:
The problem is this: Imagine the user puts in a format specifier %.2f so 2 digits. then, as initial value he gives 100, its printed as 100.00. when you now try to edit the number, say delete a digit, after you delete, the value is parsed(its still 100) and on the next call, its printed out as 100.00 again. so you can never delete a digit

Reply:
But as per my understanding, that's right, depending on the display format edition will require changing more digits from the text but that text is processed after editing

Reply:
ok, i get what you mean, if the text is only parsed after the valuebox loses focus, it should work

Reply:
Hey again, 
after taking a look again i dont think its possible without some sort of text buffer thats persistent between calls. any time the function is called, the float value provided will be converted to text. its impossible to delete the decimal digits because in the next frame they will just be printed out again.

Reply:
> But as per my understanding, that's right, depending on the display format, edition will require changing more digits from the text but that text is processed after editing

the text is always processed while editing, any time you change a digit the text is parsed

Reply:
Then why not use a small static text buffer? `static char text[32]`. It can be only used on editing and avoid processing the float from it until exiting editing mode. Actually, the only limitation of this implementation would be that 2 `valuebox` can not be editable at the same time.

Function can also be called `GuiValueBoxEx()` as an extended value box with float and formatting support (even for integer values).

Reply:
I tried that right now. With a single Valuebox it works fine, but if you have multiple valueboxes the problem is then that every change to the text of one box is overwritten to the other boxes as well, since they have the same text buffer.